AirBoard/AirBoard/View Model/AirportViewModel.swift

```diff
@@ -17,85 +17,88 @@ class AirportViewModel {
     // MARK: Properties
     weak var delegate: AirportsViewModelDelegate?
     
-    private var airports: [Airport]
-    var airportsToDisplay: [String: [Airport]]
-    var sectionTitles: [String]
+    var data = [String: [Airport]]() {
+        didSet {
+            delegate?.reciveData()
+        }
+    }
+    var sectionTitles: [String] {
+        get {
+            return data.keys.sorted()
+        }
+    }
     
-    private let airportService: AirportService
+    private var airports = [Airport]()
+    private let airportService = AirportService()
     private let coreDataManager: CoreDataManager
     
     init(appDelegate: AppDelegate) {
         coreDataManager = CoreDataManager(appDelegate: appDelegate)
-        airportService = AirportService()
-        airports = []
-        airportsToDisplay = [:]
-        sectionTitles = []
     }
     
     func getData() {
-        if airports.isEmpty {
+        if data.isEmpty {
             loadDataFromDataBase()
         }
     }
     
     func searchAirports(cityName: String) {
         if cityName.isEmpty {
-            airportsToDisplay.removeAll()
-            prepareToDisplay()
+            data.removeAll()
         } else {
             let filteredAirports = airports.filter { $0.name.lowercased().contains(cityName.lowercased()) }
-            airportsToDisplay = ["Founded": filteredAirports]
-            sectionTitles = ["Founded"]
+            data = ["Founded": filteredAirports]
         }
-        
-        delegate?.reciveData()
     }
     
     // MARK: Private methods
     
     private func loadDataFromDataBase() {
         coreDataManager.loadAirportsFromDB(success: { [weak self] data in
-            self?.airports = data
-            self?.prepareToDisplay()
-            self?.delegate?.reciveData()
+            guard let self = self else {
+                return
```

Comment:
this construction allow to use self in safe way without optional.
https://benscheirman.com/2018/09/capturing-self-with-swift-4-2/